akvo/rsr/filters.py

```diff
@@ -98,26 +105,47 @@ def get_locations(location, locations):
 def location_choices(qs):
```

Comment:
This function is very hard to understand in general. I think it's part of the old code base, right?